x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/CrossClusterAccessSubjectInfo.java

```diff
@@ -37,7 +42,22 @@
 import java.util.Set;
 
 public final class CrossClusterAccessSubjectInfo {
+
     public static final String CROSS_CLUSTER_ACCESS_SUBJECT_INFO_HEADER_KEY = "_cross_cluster_access_subject_info";
+    private static final Logger logger = LogManager.getLogger(CrossClusterAccessSubjectInfo.class);
+    private static final Set<String> API_KEY_AUTHENTICATION_METADATA_TO_KEEP = Set.of(
+        AuthenticationField.API_KEY_ID_KEY,
```

Comment:
I've included these here since they cross-cluster-access specific: for example, the fact that we keep some of the API-key metadata fields but not other only makes sense in the RCS context.